packages/ui/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,78 @@
+import type { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import { forwardRef } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { Icon } from './Icon';
+import type { BaseProps } from './types';
+
+export interface ButtonProps extends BaseProps, ButtonHTMLAttributes<HTMLButtonElement> {
+    isLoading?: boolean;
+    isDisabled?: boolean;
```

Comment:
There's already a `disabled` property on HTMLButton, let's re-use that.
Same for `onClick`.